tenable/io/v3/vm/agent_groups/api.py

```diff
@@ -0,0 +1,215 @@
+
+Methods available on ``tio.v3.vm.agent_groups``:
+
+.. rst-class:: hide-signature
+.. autoclass:: AgentGroupsAPI
+    :members:
+'''
+from typing import Dict, Union
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.vm.agent_groups.schema import AgentGroupsBaseSchema
+
+
+class AgentGroupsAPI(ExploreBaseEndpoint):
+    _path: str = f'api/v3/agent-groups'
+    _conv_json: bool = True
+
+    def add_agent(self, group_id: UUID, *agent_ids: UUID) -> Union[None, Dict]:
+        '''
+        Adds an agent or multiple agents to the agent group specified.
+
+        :devportal:`agent-groups: add-agent <agent-groups-add-agent>`
+
+        Args:
+            group_id (UUID): The unique identifier of the group
+            agent_ids (UUID): The unique identifier of the agent
+
+        Returns:
+            :obj:`dict` or :obj:`None`:
+                If adding a singular agent, a :obj:`None` response will be
+                returned.  If adding multiple agents, a :obj:`dict` response
+                will be returned with a task record.
+
+        Examples:
+            Adding a singular agent:
+
+            >>> tio.v3.vm.agent_groups.add_agent(
+            ...    'ef62870e-fe2f-4ba9-98b7-43d3a53ffe85',
+            ...    '1bd703af-b2aa-4a82-ad8d-b883381a873f'
+            ... )
+
+            Adding multiple agents:
+
+            >>> tio.v3.vm.agent_groups.add_agent(
+            ...    'ef62870e-fe2f-4ba9-98b7-43d3a53ffe85',
+            ...    '1drge3af-b2aa-4a81-ad8d-b883381a873f',
+            ...    '1bgfdgaf-b2aa-4a82-ad8d-b834581a873f',
+            ...    'bsbsbbdf-b2aa-4a83-ad8d-b867581a873f'
+            ... )
+        '''
+        if len(agent_ids) <= 1:
+            # if there is only 1 agent id, we will perform a singular add.
+            self._put(f'{group_id}/agents/{agent_ids[0]}')
+        else:
+            # If there are many agent_ids, then we will want to perform a
+            # bulk operation.
+            payload: dict = {'items': [i for i in agent_ids]}
+            schema = AgentGroupsBaseSchema(only=['items'])
+            payload = schema.dump(schema.load(payload))
+            return self._post(
+                f'{group_id}/agents/_bulk/add',
+                json=payload
+            )
+
+    def configure(self, group_id: UUID, name: str) -> Dict:
+        '''
+        Renames an existing agent group.
+
+        :devportal:`agent-groups: configure <agent-groups-configure>`
+
+        Args:
+            group_id (UUID): The unique identifier of the group
+            name (str): The new name for the agent group
+
+        Returns:
+            :obj:`dict`
+
+        Examples:
+            >>> tio.v3.vm.agent_groups.configure(
+            ...    'ef62870e-fe2f-4ba9-98b7-43d3a53ffe85',
+            ...    'New Name'
+            ... )
+        '''
+        return self._put(f'{group_id}', json={'name': name})
```

Comment:
Can we validate this payload with schema as well similar to create action

Reply:
yes, we can. updated as per comment.